### PR TITLE
fix: don't attempt to login during a --dryRun

### LINKED ISSUE #1023: 🐛 BUG: Running wrangler with --dry-run shouldn't try log me in
### What version of `Wrangler` are you using?

2.0.5

### What operating system are you using?

Mac

### Describe the Bug

Running wrangler with the `--dry-run` option shouldn't try to log me in

### PR DESCRIPTION
**Pull request by rozenmd**
Closes #1023 

//TODO:

- [x] Update tests to ensure dry-run works even without an account

### CODE REVIEW

File: packages/wrangler/src/index.tsx [hunk 1]
```diff
@@ -1256,7 +1256,7 @@ export async function main(argv: string[]): Promise<void> {
         );
       }
 
-      const accountId = await requireAuth(config);
```
Comment (rozenmd) [inline packages/wrangler/src/index.tsx:1259]: @threepointone Is it an issue that during `--dry-run`, accountId won't be set even if you're logged in? We're not like generating a manifest with accountIds/paths that rely on accountId being set?

Comment (threepointone) [reply packages/wrangler/src/index.tsx:1259]: Well we want zero requests during dry run, so maybe it's good that we enforce it doubly here. We'll want to generate a manifest, but we probably won't include account id in it (yet).

**Review by threepointone — APPROVED** [review]
Looks good!

**Review by petebacondarwin — APPROVED** [review]

**Comment by petebacondarwin** [comment]
FWIW I think there is potential for there to be two kinds of "dry-run". One is what we are implementing here, which is where we promise never to make any network requests at all. The other is where we promise not to do any network requests that are not readonly. The latter could be useful since we could literally "dry-run" every aspect of the logic, including doing verifications that require API interactions, such as listing off which assets would need uploading.

**Pull request merged**